cassandra-protocol/src/types/vector.rs

```diff
@@ -0,0 +1,76 @@
+use crate::error::Result;
+use crate::frame::message_result::{ColType, ColTypeOption, ColTypeOptionValue};
+use crate::frame::Version;
+use crate::types::data_serialization_types::*;
+use crate::types::{AsRust, AsRustType, CBytes};
+use derive_more::Constructor;
+
+// TODO: consider using pointers to ColTypeOption and Vec<CBytes> instead of owning them.
+#[derive(Debug, Constructor)]
+pub struct Vector {
+    /// column spec of the list, i.e. id should be List as it's a list and value should contain
+    /// a type of list items.
+    metadata: ColTypeOption,
+    data: Vec<CBytes>,
+    protocol_version: Version,
+}
+
+impl Vector {
+    fn map<T, F>(&self, f: F) -> Vec<T>
+    where
+        F: FnMut(&CBytes) -> T,
+    {
+        self.data.iter().map(f).collect()
+    }
+}
+
+pub struct VectorInfo {
+    pub internal_type: String,
+    pub count: usize,
+}
+
+pub fn get_vector_type_info(option_value: &ColTypeOptionValue) -> Result<VectorInfo> {
```

Comment:
Please don't use panicking functions or panic itself, since we're already using a Result.